.github/workflows/publish.yml

```
@@ -0,0 +1,32 @@
+name: Publish
+on:
+  push:
+    tags:
+      - "v0.*"
+jobs:
+  publish:
+    environment:
+      name: maven-central-and-ruby-gems
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up OpenJDK 8
+      uses: actions/setup-java@v4
+      with:
+        java-version: 8
+        distribution: "temurin"
+    - name: Publish
+      run: |
+        mkdir -p $HOME/.gem
+        touch $HOME/.gem/credentials
+        chmod 0600 $HOME/.gem/credentials
+        printf -- "---\n:rubygems_api_key: ${RUBYGEMS_API_KEY}\n" > $HOME/.gem/credentials
+        ./gradlew --stacktrace publishMavenPublicationToMavenCentralRepository gemPush
+      env:
+        ORG_GRADLE_PROJECT_ossrhUsername: ${{ secrets.OSSRH_USERNAME }}
```

We use `vars`, instead of `secrets`, for username, because the username is not a secret.

```suggestion
        ORG_GRADLE_PROJECT_ossrhUsername: ${{ vars.OSSRH_USERNAME }}
```

You can see this configuration at:
* "Settings" of this repository
* → "Environments"
* → `maven-central-and-ruby-gems`
* → Environment secrets and Environment variables

---

Thanks! fixed. I copied https://github.com/embulk/embulk-input-s3/blob/master/.github/workflows/publish.yml

---

Yes, some older ones are yet like that. (GitHub only provided secrets in earlier days.)